frontend/app/controllers/PayPal.scala

```diff
@@ -18,105 +14,45 @@ object PayPal extends Controller with LazyLogging {
 	implicit val tokenWrites = Json.writes[Token]
 	implicit val tokenReads = Json.reads[Token]
 
-	// The parameters sent with every NVP request.
-	private val defaultNVPParams = Map(
-		"USER" -> Config.paypalUser,
-		"PWD" -> Config.paypalPassword,
-		"SIGNATURE" -> Config.paypalSignature,
-		"VERSION" -> Config.paypalNVPVersion)
-
-	// Takes a series of parameters, send a request to PayPal, returns response.
-	private def nvpRequest (params: Map[String, String]) = {
-
-		val client = new OkHttpClient()
-		val reqBody = new FormBody.Builder()
-		for ((param, value) <- defaultNVPParams) reqBody.add(param, value)
-		for ((param, value) <- params) reqBody.add(param, value)
-
-		val request = new Request.Builder()
-			.url(Config.paypalUrl)
-			.post(reqBody.build())
-			.build()
-
-		client.newCall(request).execute()
-
-	}
-
-	// Takes an NVP response and retrieves a given parameter as a string.
-	private def retrieveNVPParam (response: Response, paramName: String) = {
-
-		val responseBody = response.body().string()
-		val queryParams = parseQuery(responseBody)
-		queryParams.paramMap(paramName).head
-
-	}
-
-	// Retrieves a payment token from an NVP response, and wraps it in JSON for
-	// sending back to the client.
-	private def tokenJsonResponse (response: Response) = {
-
-		val token = Token(retrieveNVPParam(response, "TOKEN"))
-		Json.toJson(token)
-
-	}
-
-	// Sends a request to PayPal to create billing agreement and returns BAID.
-	private def retrieveBaid (token: Token) = {
-    logger.info("Called retrieveBaid")
-		val agreementParams = Map(
-			"METHOD" -> "CreateBillingAgreement",
-			"TOKEN" -> token.token)
-
-		val response = nvpRequest(agreementParams)
-		Json.toJson(Token(retrieveNVPParam(response, "BILLINGAGREEMENTID")))
-
+	// Wraps the PayPal token and converts it to JSON
+  // for sending back to the client.
+	private def tokenJsonResponse (token : String) = {
```

Comment:
I know we discussed this when we first implemented the PayPal Controller, but I find this conceptually a little bit confusing. We're using this to serialise both the `token` and the `baid`, but referring throughout to it just as the token. I'm just wondering whether we should name things slightly more generically, or possibly use a separate method for the `baid`.

Obvs this is just a readability thing, functionally it's fine. What do you think @rupertbates?